django/core/management/commands/makemessages.py

```diff
@@ -153,16 +153,45 @@ def cleanup(self):
                 os.unlink(self.work_path)
 
 
+def build_text_file_content(lines_list=None, raw_contents=None):
+    """
+    Takes either a list of lines without EOL sequences (possibly already
+    returned by str.splitlines()) or a block of raw text that will be passed
+    through str.splitlines() to get Universal newlines treatment (has
+    precedence).
+
+    Returns the resulting block of text with normalised `\n` EOL sequences
+    ready to be written to disk using current platform native EOLs.
+    """
+    assert bool(lines_list is not None) != bool(raw_contents is not None)
```

Comment:
You've made me realize it doesn't need to handle both cases. We can handle one caling `\n.join(...)`.

I've simplified and renamed the function.